### PR TITLE
fix: issue with empty search results in listbox

### PR DESCRIPTION
**Pull request by StefanEnberg**
## Motivation

Made this fix very explicit so that it is not hard to miss (+1 on line 222 would have looked slicker)

## Requirements checklist

- [ ] Api specification
  - [ ] Ran `yarn spec`
    - [ ] No changes
     ***OR***
    - [ ] API changes has been formally approved
- [ ] Unit/Component test coverage
- [ ] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:
- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 1]
```diff
@@ -196,7 +196,7 @@ export default function ListBox({
     if (typeof setCount === 'function' && layout) {
       setCount(layout.qListObject.qSize.qcy);
     }
-  }, [layout]);
+  }, [layout, layout && layout.qListObject.qSize.qcy]);
```
Comment (veinfors) [inline apis/nucleus/src/components/listbox/ListBox.jsx:199]: This diff is confusing me a little... this is actually what it looks like on master.. maybe something is messed up with this branch or the PR. It's probably fine but a little confusing that we see this diff.

Comment (veinfors) [inline apis/nucleus/src/components/listbox/ListBox.jsx:199]: This diff is confusing me a little... this is actually what it looks like on master.. maybe something is messed up with this branch or the PR. It's probably fine but a little confusing that we see this diff.

File: apis/nucleus/src/components/listbox/ListBox.jsx [hunk 2]
```diff
@@ -218,7 +218,14 @@ export default function ListBox({
   }
 
   const isVertical = listLayout !== 'horizontal';
-  const count = layout.qListObject.qSize.qcy;
+
+  let count = layout.qListObject.qSize.qcy;
+
+  // Should not allow a count of zero since it will stop executing loadMoreItems
+  // this could happen if a seach returns an empty result
+  if (count === 0) {
```
Comment (veinfors) [inline apis/nucleus/src/components/listbox/ListBox.jsx:226]: I tested this and it seems to work as expected and solve the problem!
Would be nice with a unit test covering this though.

**Review by veinfors — APPROVED** [review]

File: apis/nucleus/src/components/listbox/ListBox.jsx (hunk 2 above)
Comment (StefanEnberg) [reply apis/nucleus/src/components/listbox/ListBox.jsx:226]: https://github.com/qlik-oss/nebula.js/pull/886

**Pull request merged**